src/remediations/fifi.js

```diff
@@ -46,7 +46,18 @@ async function fetchRHCClientId (systems) {
     const systemIds = _.map(systems, system => { return system.id; });
     const systemProfileDetails = await inventory.getSystemProfileBatch(systemIds);
 
-    _.forEach(systems, system => system.rhc_client = systemProfileDetails[system.id].system_profile.rhc_client_id);
+    _.forEach(systems, system => {
+        system.rhc_client = systemProfileDetails[system.id].system_profile.rhc_client_id;
+        system.marketplace = systemProfileDetails[system.id].system_profile.is_marketplace;
+    });
+}
+
+function sortRHCSystems (executor, smart_management) {
+    if (smart_management) {
+        return _.partition(executor.systems, system => !_.isUndefined(system.rhc_client));
+    }
+
+    return _.partition(executor.systems, system => !_.isUndefined(system.rhc_client) && system.marketplace);
```

Comment:
It may be worth making a distinction between hosts that don't have RHC set up and those that cannot be run due to missing smart_management. So that the user knows what they are missing.